Source/Protected.swift

```diff
@@ -24,10 +24,65 @@
 
 import Foundation
 
-// MARK: -
+private protocol Lock {
+    func lock()
+    func unlock()
+}
+
+extension Lock {
+    /// Executes a closure returning a value while acquiring the lock.
+    ///
+    /// - Parameter closure: The closure to run.
+    ///
+    /// - Returns:           The value the closure generated.
+    func around<T>(_ closure: () -> T) -> T {
+        lock(); defer { unlock() }
+        return closure()
+    }
+
+    /// Execute a closure while acquiring the lock.
+    ///
+    /// - Parameter closure: The closure to run.
+    func around(_ closure: () -> Void) {
```

Comment:
Is there a reason to include this version? The project builds and the tests pass without it; I’m assuming the compiler can infer `T` as `Void` if it needs to.

Reply:
My only thought is that the unspecialized version may optimize where the generic fails. Otherwise you're right, they're functionally identical AFAICT. I'll leave it for now pending investigation.